### PR TITLE
BWS-32: + added common component table

### PR DESCRIPTION
**Pull request by Donymak**

### CODE REVIEW

File: frontend/src/components/common/table/table.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+const Table: React.FC<Props> = ({ columns, data }) => {
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
+    useTable({ columns, data });
+
+  return (
+    <table {...getTableProps()} className={styles.clientSideTable}>
+      <thead>
+        {headerGroups.map((headerGroup) => (
```
Comment (Oleg-feriss) [inline frontend/src/components/common/table/table.tsx]: Maybe it's better to use **key** prop.

File: frontend/package.json [hunk 2]
```diff
@@ -24,12 +24,14 @@
     "react-hook-form": "7.24.0",
     "react-redux": "7.2.6",
     "react-router-dom": "6.2.1",
-    "react-scripts": "5.0.0"
+    "react-scripts": "5.0.0",
+    "react-table": "^7.7.0"
```
Comment (what1s1ove) [inline frontend/package.json]: Check `B19` criteria

File: frontend/package.json [hunk 3]
```diff
@@ -24,12 +24,14 @@
...
   },
   "devDependencies": {
     "@types/node": "16.11.19",
     "@types/react": "17.0.38",
     "@types/react-dom": "17.0.11",
+    "@types/react-table": "^7.7.9",
```
Comment (what1s1ove) [inline frontend/package.json]: Check `B19` criteria

File: frontend/src/components/common/table/table.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  Header: string;
+  accessor: string;
+}
+
+interface ISchemaItems extends Array<ISchemaItem> {}
+
+type Props = {
+  columns: ISchemaItems;
+  data: Record<string, unknown>[];
+};
```
Comment (what1s1ove) [inline frontend/src/components/common/table/table.tsx]:
```suggestion
import { useTable, Column } from 'react-table';

type Props = {
  columns: Column[];
  data: unknown[];
};
```

File: frontend/src/components/common/table/table.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+const Table: React.FC<Props> = ({ columns, data }) => {
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
+    useTable({ columns, data });
+
+  return (
+    <table {...getTableProps()} className={styles.clientSideTable}>
```
Comment (what1s1ove) [inline frontend/src/components/common/table/table.tsx]: what about table title from the markup?

File: frontend/src/components/common/table/table.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+interface ISchemaItems extends Array<ISchemaItem> {}
+
+type Props = {
+  columns: ISchemaItems;
+  data: Record<string, unknown>[];
+};
+
+const Table: React.FC<Props> = ({ columns, data }) => {
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
+    useTable({ columns, data });
```
Comment (what1s1ove) [inline frontend/src/components/common/table/table.tsx]:
```suggestion
    useTable({
    columns: columns as Column<Record<string, string>>[],
    data: data as Record<string, string>[],
    });
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/table/table.module.scss [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+.tableRow:hover {
+  background-color: #2a2a2a82;
+  cursor: pointer;
+  transition-duration: 500ms;
+}
+
+.tableCell {
+  color: #ffffff;
+}
```
Comment (Yaroslaww-1) [inline frontend/src/components/common/table/table.module.scss]: I think its better to use styles nesting here for visibility

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/table/table.module.scss (hunk 7 above)
Comment (Donymak) [reply frontend/src/components/common/table/table.module.scss]: Eslint: Expected nesting length to be no more than 0

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**